src/Airship/Resource.hs

```diff
@@ -22,57 +22,100 @@ import Blaze.ByteString.Builder.Html.Utf8 (fromHtmlEscapedText)
 import Network.HTTP.Types
 import Network.HTTP.Media (MediaType)
 
+-- | Used when processing POST requests so as to handle the outcome of the binary decisions between
+-- handling a POST as a create request and whether to redirect after the POST is done.
 -- Credit for this idea goes to Richard Wallace (purefn) on Webcrank.
--- This creates an enumeration for the four possibilities of two binary
--- decisions:
--- postIsCreate = True | False
--- redirect     = True | False
 data PostResponse s m
-    = PostCreate [Text] -- ^ Treat this request as a PUT
-    | PostCreateRedirect [Text] -- ^ Treat this request as a PUT, then redirect
-    | PostProcess (Handler s m ()) -- ^ Process, but don't redirect
-    | PostProcessRedirect (Handler s m ByteString)
+    = PostCreate [Text] -- ^ Treat this request as a PUT.
+    | PostCreateRedirect [Text] -- ^ Treat this request as a PUT, then redirect.
+    | PostProcess (Handler s m ()) -- ^ Process as a POST, but don't redirect.
+    | PostProcessRedirect (Handler s m ByteString) -- ^ Process and redirect.
 
 data Resource s m =
-    Resource { allowMissingPost         :: Handler s m Bool
+    Resource { -- | Whether to allow HTTP POSTs to a missing resource. Default: false.
+               allowMissingPost         :: Handler s m Bool
+               -- | The set of HTTP methods that this resource allows. Default: @GET@ and @HEAD@.
```

Comment:
Yep. Tentatively implemented [like in Erlang Webmachine](https://github.com/basho/webmachine/wiki/Resource-Functions#options). Something you might be interested in working on, or just curious?